### ISSUE TITLE
Abbreviated formal parameter lists

### ISSUE THREAD

**Issue by eernstg** [dart-lang/language#57]
Setting out from the discussion 'Constructor inheritance and named arguments' (started by Mouad Debbar) on dart-language-discuss@google.com, there is a need for abbreviation of formal parameter lists in the cases where they are determined by the context. In particular, there is a lot of redundancy in the case where a long list of parameters with verbose types are just passed on from a constructor to another constructor in a superinitialization, and similar situations arise with forwarding functions:

```dart
// Example from Mouad Debbar.

class Base {
  final int foo;
  final String bar;
  final bool baz;
  Base({this.foo, this.bar, this.baz});
}

class Sub1 extends Base {
  final List<int> myFoo;
  Sub1({this.myFoo, int foo, String bar, bool baz}): super(foo: foo, bar: bar, baz: baz);
}

class Sub2 extends Sub1 {
  final Map<String, int> myBar;
  Sub2({this.myBar, List<int> myFoo, int foo, String bar, bool baz}):
      super(myFoo: myFoo, foo: foo, bar: bar, baz: baz);
}
```
Samual Rawlins added the following concerns: It is at times necessary to import additional libraries, just so we can talk about the redundant elements in such formal parameter lists and in their use for a forwarding call, and we may also have to repeat some default values (and get them right).

**Comment by yjbanov** [comment]
When thinking about solutions to this problem, please also consider keeping the forwarding issue orthogonal to the default value issue.

**Scenario**: on day 1 the author of `Sub1` was 100% satisfied forwarding one of the constructors to a `super` constructor. The forwarding method allowed to inherit not only the constructor signature, but also all the parameter defaults. On day 2, the author of `Sub1` decided to customize the constructor in some way, and stopped using the forwarding mechanism. Unfortunately, that also reset all the default values. Even more unfortunate, the default values on the `super` constructor turned out to be _private_. The author of `Sub1` is now stuck :sob:

**Comment by mdebbar** [comment]
@yjbanov agreed. I think we should allow forwarding methods/constructors to have bodies. That's possible with the approach explained in https://github.com/dart-lang/language/issues/58#issuecomment-432738770.
